# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: tools/profiler/public/GeckoProfiler.h
# Commit: 3aa39fa769a2
# Full Hash: 3aa39fa769a237babc6d2a09c39a4de58e9e589b
# Author: Gerald Squelart <gsquelart@mozilla.com>
# Date: 2020-10-22 21:51:59
# Description:
#   Bug 1672501 - Initialize scProfilerMainThreadId from profiler_init - r=gregtatum
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D94419
# ==============================================================================

diff -r 766437adb95f -r 3aa39fa769a2 tools/profiler/public/GeckoProfiler.h
--- a/tools/profiler/public/GeckoProfiler.h	Thu Oct 22 12:46:28 2020 +0000
+++ b/tools/profiler/public/GeckoProfiler.h	Thu Oct 22 12:55:13 2020 +0000
@@ -640,7 +640,9 @@
 // Get the current thread's ID.
 int profiler_current_thread_id();
 
-extern const int scProfilerMainThreadId;
+// Statically initialized to 0, then set once from profiler_init(), which should
+// be called from the main thread before any other use of the profiler.
+extern int scProfilerMainThreadId;
 
 inline int profiler_main_thread_id() { return scProfilerMainThreadId; }
 
